from-repo/c-cpp/build-generator.py

```diff
@@ -0,0 +1,1341 @@
+#!/usr/bin/env python3
```

Comment:
Can you add some docs (e.g. a markdown doc) explaining the high level design ideas / flow for how this works?

Reply:
Yes, done